spirv_msl.hpp

```diff
@@ -144,10 +147,11 @@ class CompilerMSL : public CompilerGLSL
 	uint32_t get_ordered_member_location(uint32_t type_id, uint32_t index);
 	uint32_t pad_to_offset(SPIRType &struct_type, bool is_indxd_vtx_input, uint32_t offset, uint32_t struct_size);
 	SPIRType &get_pad_type(uint32_t pad_len);
-	size_t get_declared_type_size(const SPIRType &type) const;
-	size_t get_declared_type_size(const SPIRType &type, uint64_t dec_mask) const;
+	size_t get_declared_type_size(uint32_t type_id) const;
+	size_t get_declared_type_size(uint32_t type_id, uint64_t dec_mask) const;
 
 	MSLConfiguration msl_config;
+	std::unordered_set<uint32_t> custom_function_ops;
```

Comment:
Changed to use std::set instead of std::unordered_set, to retain ordering...but also only record the need for each function only once. Using std::vector results in the potential output of multiple copies of the declaration and definition of a custom function.